### PR TITLE
Support Camel case URI variables

### PR DESCRIPTION
**Pull request by rwinch**
Perviously there were issues with case insenstive patterns and URI
variables that contained upper case characters. For example, the pattern
"/user/{userId}" could not resolve the variable #userId Instead it was
forced to lowercase and #userid was used.

Now if the pattern is case insensitive then so is the variable. This means
that #userId will work as will #userid.

Fixes gh-3786

### CODE REVIEW

File: config/src/test/groovy/org/springframework/security/config/http/InterceptUrlConfigTests.groovy [hunk 1]
```diff
@@ -137,6 +137,39 @@ class InterceptUrlConfigTests extends AbstractHttpConfigTests {
...
 	}
 
+	def "gh-3786 intercept-url supports cammel case path variables"() {
+		setup:
+		MockHttpServletRequest request = new MockHttpServletRequest(method:'GET')
+		MockHttpServletResponse response = new MockHttpServletResponse()
+		MockFilterChain chain = new MockFilterChain()
+		xml.http('use-expressions':true) {
+			'http-basic'()
+			'intercept-url'(pattern: '/user/{userName}/**', access: "#userName == authentication.name")
```
Comment (jgrandja) [inline config/src/test/groovy/org/springframework/security/config/http/InterceptUrlConfigTests.groovy]: Should we add another test where the uri pattern variable (userName) is lowercase referenced in the access check (access: #username)

File: config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeRequestsTests.java [hunk 2]
```diff
@@ -157,6 +157,47 @@ protected void configure(AuthenticationManagerBuilder auth) throws Exception {
 		}
 	}
 
+	// gh-3786
+	@Test
+	public void antMatchersPathVariablesCaseInsensitiveCamelCaseVariables() throws Exception {
+		loadConfig(AntPatchersPathVariablesCammelCaseVariables.class);
```
Comment (jgrandja) [inline config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeRequestsTests.java]: Typo: should be Ant**M**atchersPathVariablesCammelCaseVariables

File: config/src/test/groovy/org/springframework/security/config/http/InterceptUrlConfigTests.groovy (hunk 1 above)
Comment (rwinch) [inline config/src/test/groovy/org/springframework/security/config/http/InterceptUrlConfigTests.groovy]: This is more of a integration test. The unit tesst cover the edge cases

File: config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeRequestsTests.java (hunk 2 above)
Comment (rwinch) [inline config/src/test/java/org/springframework/security/config/annotation/web/configurers/AuthorizeRequestsTests.java]: Thanks! I pushed a fix

**Comment by jgrandja** [comment]
The for the PR @rwinch!
